rts/Game/GameHelper.cpp

```diff
@@ -954,7 +957,12 @@ void CGameHelper::BuggerOffRectangle(const float3& mins, const float3& maxs, boo
 			continue;
 
 		const float footPrintRadius = math::sqrtf(footPrintX*footPrintX + footPrintZ*footPrintZ);
-		const float buggerOffRadius = baseBuggerOffRadius + footPrintRadius;
+
+		// If the unit radius is smaller than the moveDef foot print then the bugger off may not be honoured by the
+		// unit because the bugger off command uses unit radius to determine whether the unit has moved far enough.
+		const float addForUnitRadius = (footPrintRadius > u->radius) ? (footPrintRadius - u->radius) : 0.f;
```

Comment:
Why not just `std::max(footprintRadius, u->radius)`?

Reply:
Because we need to add increase the buggerOffRadius if footprintRadius is bigger than u->radius, which isn't a max function. The amount we add is based on how much bigger footprintRadius is to u->radius. The comment explains why we need to do this.

Reply:
I see, I misread the formula. But now the change looks even more confusing. Here's a top-down illustration for how far away a unit should go to stop blocking the bugger off area, assuming footprint > radius. This corresponds to how the formula looked before the change, i.e. just `base + movedef footprint`:
![image](https://github.com/user-attachments/assets/1390b8d2-1e5c-417e-85c6-bf9ed9b4be8b)

Now, before the `unit->footprint := movedef->footprint` change that formula could fail in this way when the footprints weren't equal, where the unit footprint would still cover part of the bugger-off area and prevent construction:
![image](https://github.com/user-attachments/assets/f8154309-be30-44b5-93f8-18236d757958)

But now that both footprints are equalized this shouldn't be possible anymore. So this change looks redundant.

Reply:
A) So unit->footprint is used to do a square-by-square analysis of whether a unit blocks the placement of a building - that's because the unit->footprint is used to fill the ground blocking map, which is what is looked up square by square.

B) Bugger off is handled in two parts.
1) checking whether a unit is overlaping the building rectangle which was calculated just by looking at the movedef->footprint. If so, then the unit is issued a bugger-off command with a bugger off radius (using the building's footprint radius and radius of the obsctructing unit's moveDef->footprint.)
2) processing bugger off commands. The units checks whether its radius overlaps the the bugger off radius.

Now if the unit's radius is smaller than its movedef->footprint, then the unit could think it is outside the bugger off radius and won't try to move, despite overlapping the build area, which is why now step B.1 extends the bugger off radius sent to the unit to account for the difference between the unit's radius and the movedef.

So the bugger off radius should be: `max( moveDef->footprint, moveDef->footprint + (moveDef->footprint - u->radius) )`

You can think of it as another problem similar to the unit/movedef foorptint issue, just for moveDef vs unit->radius.

I suppose one could argue that a larger unit radius would suggest we could shortern the buggeroff radius. That could work. It would simplify the code a little and make it easier to follow. So we could have `moveDef->footprint + (moveDef->footprint - u->radius)`?

Reply:
> if the unit's radius is smaller than its movedef->footprint, then the unit could think it is outside the bugger off radius and won't try to move, despite overlapping the build area

I don't believe this is actually possible after the changes in `GroundMoveType.cpp`. Can you draw a simple counterexample picture in mspaint like above?

Reply:
Actually, no. You are correct. I will remove that bit.